build/scripts/previousversion/main.go

```diff
@@ -0,0 +1,16 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+
+	semver "github.com/blang/semver/v4"
+)
+
+func main() {
+	baseVersion := flag.String("version", "", "the base version to be taken in as input for release")
+	flag.Parse()
+	v, _ := semver.Parse(*baseVersion)
+	releaseVersion := fmt.Sprintf("%d.%d.%d", v.Major, v.Minor-1, v.Patch)
+	fmt.Println(releaseVersion)
+}
```

Comment:
🤔 do we put a small go.mod in this directory, rather than add at the root level (then we also don't need to vendor the code either)

Since this gets run once every 6 weeks, it probably doesn't need to be vendor ed. WDYT?

Reply:
Hi, I believe that creating multiple go.mod might require additional efforts on maintaining the documentation and explaining the reason behind adding it seperately along with the major vendor file. Also, creating multiple such directories and executables might lead to us maintaining multiple go.mod dependencies, which i think might be confusing. Just my thought.

Reply:
I'd probably argue that it's easier to manage and maintain to keep it seperate. This go.mod is for development tooling - whereas the root go.mod is for building production Agones.

🤔 maybe a good in-between would be to have a go.mod in `build/scripts` - then we can have a central place for any build-tooling Go dependencies, but they can be kept separate from the dependencies and Go version for the actual Agones code. How does that sound?

Reply:
Fair point, will try this.